cha/external_architecture.tex

```diff
@@ -0,0 +1,47 @@
+Furthermore the server would act as an authority, providing a clock to synchronize towards.
+
+The solution also relies on an internet connection being available and is affected by bandwidth, furthermore a server needs to be available, in the case of widespread use we would need enough hardware to support multiple users. 
+Lastly the server would also provide a single point of failure; no server, no application.
+
+
+\subsection{Local Centralized}
+While having external hardware available and requiring an internet connection would limit the application use scenarios, this can be rectified by localizing the solution.
+To do so a device would act as the server, a master device.
+The session host would become the master, and any connecting devices would be slaves.
+A master device would, similarly to the server, provide an authority to synchronize towards.
+Slaves would retain the status which android devices holds in the remote centralized solution, simply acting as information providers and speakers, whereas the master would manage computational requirements, LAN management and file distribution.
+
+In contrast to the remote centralized idea this would remove the need for external hardware and decrease latency however, we also produce the need for peer discovery and LAN management.
+We would still have the single point of failure in the master, but with no internet requirement.
+
+\subsection{Peer--To--Peer}
+A common denominator between the two previous solutions is their single point of failure produced by their master--slave type relationship.
+A Peer--To--Peer solution would remove this single point of failure, but not without adding its own complications.
+In a Peer--To--Peer solution all devices would be equal, this in turn means that should a device fail or leave the session, it does not matter which device it was as the network of devices would be unaffected, given that no device is of higher authority.
+With all devices being equal it also means having to reach consensus to make decisions, simple decisions like playback control could be managed through a majority voting system, but the more complex underlying communication such as synchronization is infinitely more complex due to not having an authority.
+
+This solution could be expanded to consider a rotating master which would help resolve possible consensus issues, but at that point it becomes overcomplicating the local centralized solution in order to reduce the chance of the master leaving or failing, however that would still be possible.
+
+
+\subsection{Conclusion}
+Ultimately the key challenge is synchronizing the clocks, and to utilize a synchronized network even a few milliseconds matter; as such we are not interested in a Peer--To--Peer network as synchronization in a Peer-To--Network makes this challenge far more complex than it already is.

```

Comment:
Synes at "synchronizing the clocks" kommer lidt pludselig her som et stort problem. Det er ikke nævnt under Local Centralized, og i Remote Centralized er det mere som noget der kan gøres med serveren, det er kun ved P2P, men det er mere som P2P der gør det svært. Måske lidt mere fokus i de tre sync metoder at sync af clocks er et problem, så bliver det nemmere at retfærdiggøre senere hvorfor vi bruger energi på det.